custom/icds_reports/sqldata/exports/growth_tracker_report.py

```diff
@@ -0,0 +1,178 @@
+from corehq.apps.locations.models import SQLLocation
+from custom.icds_reports.models import ChildHealthMonthlyView
+from custom.icds_reports.sqldata.base import IcdsSqlData
+from custom.icds_reports.utils.mixins import ExportableMixin
+from custom.icds_reports.utils import get_status, calculate_date_for_age, \
+    current_month_stunting_column, \
+    current_month_wasting_column, phone_number_function, india_now
+from dateutil.relativedelta import relativedelta
+
+
+class GrowthTrackerExport(ExportableMixin, IcdsSqlData):
+    title = 'Child Growth Tracking'
+    table_name = 'child_health_monthly_view'
+
+    def get_excel_data(self, location):
+
+        def _check_case_presence(case_id, column, data_dict):
+            return data_dict[case_id][column] if case_id in data_dict.keys() else "N/A"
+
+        def _fetch_data(filters, order_by, case_by_grouping=False):
+            query_set = ChildHealthMonthlyView.objects.filter(**filters).order_by(*order_by)
+            data_month = query_set.values('state_name', 'district_name', 'block_name', 'supervisor_name',
+                                          'awc_name', 'awc_site_code', 'person_name', 'dob', 'mother_name',
+                                          'mother_phone_number', 'pse_days_attended', 'lunch_count',
+                                          'current_month_nutrition_status',
+                                          current_month_wasting_column(self.beta),
+                                          current_month_stunting_column(self.beta), 'case_id')
+            if case_by_grouping is True:
+                data_month = {item['case_id']: item for item in data_month}
+            return data_month
+
+        filters = {
+            "month": self.config['month'],
+            "age_in_months__lte": 60,
```

Comment:
hey again, Why is there an upper bound of 5 years here. because PSE and SNP are given until 6 years?

Reply:
updated